### DISCUSSION TITLE
Did low version Monterey(12.5) can import hight OS version Image Sonoma?

### DISCUSSION THREAD

**Discussion by alexxxcs1** [cirruslabs/tart#639] [Q&A] [answered]
i use tart(2.0) create a sonoma image on macos 14.0, and use tart(1.5.0) to export image and push to my own cloud
there is low OS version Monterey download my sonoma image, and use tart(1.5.0) to import ,bu get error like this

```
 tart import ./image.tvm sonoma_xcode_15
# importing...
# zsh: bus error  tart import ./image.tvm sonoma_xcode_15
```
or

```
 tart import ./image.tvm sonoma_xcode_15
# importing...
# zsh: abort      tart import ./image.tvm sonoma_xcode_15
```

and when this error occurs, the running image will immediately exit with an error

i use macOS 14 and tart(1.5.0) to import image success ,so i think maybe tart import image need hight version OS environment toolchain ? 

so did l low OS version can import and run hight OS version image ?

**Comment by fkorotkov** [comment, marked as answer]
The import/export mechanism didn't between `1.5.0` and `2.0.0`. `*.tvm` file is just an Apple archive. Do you have enough sopace on the host for both the `*.tvm` file and the unarchived VM?

**Reply by alexxxcs1** [reply]
i create a image with disk size 200GiB, i try export image and import tvm file on same device ,the device remaining disk size must be less than 200Gib , it work.
but if i send tvm file to other device who disk size less than 200Gib, it will be error 
i don't know how does tart deal with the problem of sparse files actually occupying the disk when decompressing them?

**Reply by fkorotkov** [reply]
With the same device you already have the image locally so Apple File Systems takes care of figuring out the duplication and saving up the actual physical space. On other device you'll need space for both the archive and 200GB of the image.

**Reply by alexxxcs1** [reply]
i never try small size image import and export ,i will try it .
is it certain that if the remaining disk size of the host is larger than the original size of the image, compression and decompression can run normally? 
and does a lower version of macos(12.5) not need other support to run a higher version of macos(14.0)?

**Reply by alexxxcs1** [reply]
ok,thanks! i will try make disk image smaller

**Reply by alexxxcs1** [reply]
other question, i create big size image cuz i create image by tart‘s '--from-ipsw'  flag, not use packer plugin , packer plugin will removes recovery partition.
but if dont remove recover partition, i cant patch disk size to bigger on runtime , any suggestion for this ? 
or can you tell me how to delete the recovery partition myself? I need documentation on this

**Comment by fkorotkov** [comment]
It's possible to grow the disk manually. Please check this comment https://github.com/cirruslabs/tart/issues/17#issuecomment-1439367869

**Reply by alexxxcs1** [reply]
yes, i use this way to patch my older image disk size
but right now ,i not repair disk 'Apple_APFS_Recovery' ,so [this](https://github.com/cirruslabs/tart/issues/17#issuecomment-1439367869) is not working .
i found how packer plugin repair Apple_APFS code on plugin [codefile](https://github.com/cirruslabs/packer-plugin-tart/blob/main/builder/tart/step_disk_resize.go)
is if i remove Apple_APFS_Recovery can do [this](https://github.com/cirruslabs/tart/issues/17#issuecomment-1439367869) to patch disk size ?

**Reply by alexxxcs1** [reply]
and on my new image ,i found vm's lastPartition is not Apple_APFS [Refer to this code](https://github.com/cirruslabs/packer-plugin-tart/blob/main/builder/tart/diskutil_parser.go#L15), i got Apple_APFS_Recovery, what can i do to repair disk 'Apple_APFS' ?

**Reply by alexxxcs1** [reply]
my fault . Apple_APFS is  grow the disk partition. i didnt fount how to remove Apple_APFS_Recovery on plugin code , can u give me specific code path ? [refer](https://github.com/cirruslabs/tart/discussions/631#discussioncomment-7303035)

**Reply by alexxxcs1** [reply]
what i try to do: 
1. creat a Image without remove 'Apple_APFS_Recovery'
2.Remove 'Apple_APFS_Recovery' on other host device download my Image
3.grow the disk partition on other host device
can u give me some suggestion?

**Reply by alexxxcs1** [reply]
any way , i found where plugin [remove Apple_APFS_Recover](https://github.com/cirruslabs/packer-plugin-tart/blob/c83a5d0fcc966356791fb064cbcb5337e04acb5d/builder/tart/step_disk_file_prepare.go#L55) !!!
